documentation/homework/todo-app/src/components/TasksList.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+export default class TasksList extends React.Component {
+    constructor() {
+        super();
+    }
+
+    remove(e) {
+        const task = e.target.parentNode.querySelector('span').textContent;
+        console.log(task);
+
+        this.props.removeTask(task);
+    }
+
+    render() {
+        if (this.props.tasks.length === 0) {
+            return (
+                <p>No tasks</p>
+            )
+        }
+
+        return (
+            <ul>
+                {
+                    this.props.tasks.map((elem, index) => (
+                        <li key={index}>
+                            <span>{elem}</span>
+                            <button onClick={this.remove}>X</button>
```

Comment:
You need to send the index with the `remove` like the followings `<button onClick={() => this.remove(index)}>X</button>` and use this index to remove the item directory the array.